### ISSUE TITLE
IDs not found after calling replaceChild

### ISSUE THREAD

**Issue by AmicaNoctis** [clientside/amplesdk#161]
When a subtree is replaced with a new one and this new subtree uses some of the IDs from the old one, these IDs cannot be found again with ample.getElementById().

This is due to the fact, that the new subtree is inserted first, thus registering the new IDs and then the old subtree is removed, which causes deregistration of the shared IDs.

A fix is available at https://github.com/AmicaNoctis/amplesdk/commit/d41619498ee85f2b24029ed91ce7604283154b74

Regards,

Amica

**Comment by ilinsky** [comment]
Amica, could you please test fix implemented per rev 0db2e48a53644f9f367594f149c46b63df13cd78 and re-open issue if it still exists

**Comment by AmicaNoctis** [comment]
Sergey, my test code relies on my fork with XPath support. To test your fix, I need to merge it into my code, which might take a few days, since I'm a bit busy right now. But I will test it, unless I hear from you, that you already did so.

Amica

**Comment by AmicaNoctis** [comment]
I took some minutes to merge your code into my fork. There was an error in line 202 of Element.js (see my line comment at https://github.com/clientside/amplesdk/commit/0db2e48a53644f9f367594f149c46b63df13cd78).

When I fixed this, my tests ran successfully.

**Comment by AmicaNoctis** [comment]
Okay, looks good now! ;)

**Comment by ilinsky** [comment]
Amica, thanks again for report and spotting error in fix. At the moment I am too much concetrated on https://github.com/ilinsky/xpath2.js (it is almost done!) 

Yet, please take a note at how issue was adressed: I did not apply your fix, instead I investigated why the situation was there in first place. And what was ultimately wrong - new node was first inserted, and older then removed. The two actions should have been executed in reverse order!

**Comment by AmicaNoctis** [comment]
Sergey, I saw that you wrote your own fix and I'm happy that you investigated further. Although I already pointed out that the order was wrong, I did not make this clear enough maybe. Apart from that I forgot to correct it the same way in Node.js for DocumentFragments, which you did. That's why I'm fine with your solution.

Thumbs up for your XPath progress! I'm sure it will integrate more nicely in Ample SDK than the Google-hosted WGXPath I currently use.

Amica
